### PR TITLE
Hb 06

### PR DESCRIPTION
**Pull request by vlykhvar**
1. Create models
  - ShoppingCart
  - Ticket

2. Create DAO
  - ShoppingCartDao
  - TicketDao.

3. Create service
  - ShoppingCartService

### CODE REVIEW

File: src/main/java/com/dev/theater/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl extends DaoImpl<ShoppingCart> implements ShoppingCartDao {
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from ShoppingCart sc "
+                                        + "where sc.user = :user", ShoppingCart.class)
```
Comment (gigbat) [inline src/main/java/com/dev/theater/dao/impl/ShoppingCartDaoImpl.java]: The Better variant will be using fetch
```suggestion
            return session.createQuery("from ShoppingCart sc left join fetch sc.ticketList"
                            + " where sc.user = :user", ShoppingCart.class);
```

File: src/main/java/com/dev/theater/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_card")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (gigbat) [inline src/main/java/com/dev/theater/model/ShoppingCart.java]: and here you need to remove a parameter EAGER

File: src/main/java/com/dev/theater/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_card")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (gigbat) [inline src/main/java/com/dev/theater/model/ShoppingCart.java]: If you use annotations @MapsId then you shouldn't add @GeneratedValue, Just remove it

**Review by gigbat — CHANGES_REQUESTED** [review]
Good job

**Review by gigbat — APPROVED** [review]

File: src/main/java/com/dev/theater/dao/impl/ShoppingCartDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+@Dao
+public class ShoppingCartDaoImpl extends DaoImpl<ShoppingCart> implements ShoppingCartDao {
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from ShoppingCart sc left join fetch sc.ticketList"
+                + " where sc.user = :user", ShoppingCart.class)
+                .setParameter("user", user).getSingleResult();
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/theater/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
```

File: src/main/java/com/dev/theater/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+        shoppingCartDao.update(shoppingCartDao.getByUser(user));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/theater/service/impl/ShoppingCartServiceImpl.java]: what are you doing exactly in this method?

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/service/impl/ShoppingCartServiceImpl.java]: Idk, changed

File: src/main/java/com/dev/theater/model/Ticket.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToOne;
+
+@Entity
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/theater/model/Ticket.java]: Can't we have a lot of tickets for the same movie session?

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/dao/impl/TicketDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.dev.theater.dao.impl;
+
+import com.dev.theater.dao.TicketDao;
+import com.dev.theater.library.Dao;
+import com.dev.theater.model.Ticket;
+
+@Dao
+public class TicketDaoImpl extends DaoImpl<Ticket> implements TicketDao {
+}
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/theater/dao/impl/TicketDaoImpl.java]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**